flutter/lib/src/replay/scheduled_recorder.dart

```diff
@@ -55,32 +42,57 @@ class ScheduledScreenshotRecorder extends ReplayScreenshotRecorder {
     _callback = callback;
   }
 
-  void start() {
+  void start() async {
     assert(() {
       // The following fails if callback hasn't been provided
       // in the constructor nor set with a setter.
       _callback;
       return true;
     }());
 
-    options.log(SentryLevel.debug,
-        "$logName: starting capture (${config.width}x${config.height} @ ${config.frameRate} Hz).");
+    options.log(SentryLevel.debug, "$logName: starting capture");
     _status = _Status.running;
-    _startScheduler();
+    _restartScheduler();
+  }
+
+  void onConfigurationChanged(ScheduledScreenshotRecorderConfig config) async {
+    super.config = config;
+    options.log(SentryLevel.debug,
+        "$logName: onConfigurationChanged (${config.width}x${config.height} @ ${config.frameRate} Hz).");
+
+    _restartScheduler();
   }
 
   Future<void> _stopScheduler() {
-    return _scheduler.stop();
+    return _scheduler?.stop() ?? Future.value();
   }
 
-  void _startScheduler() {
-    _scheduler.start();
+  void _restartScheduler() async {
+    await _stopScheduler();
 
-    // We need to schedule a frame because if this happens in-between user
-    // actions, there may not be any frame captured for a long time so even
-    // the IdleFrameFiller won't have anything to repeat. This would appear
-    // as if the replay was broken.
-    options.bindingUtils.instance!.ensureVisualUpdate();
+    if (super.config == null) {
+      return;
+    }
+
+    assert(config.frameRate > 0);
```

Comment:
This is kind-of ugly as `super.config` is nullable, but `config` isn't

Reply:
not sure if this is intended, otherwise we can also make `config` nullable